osmnames/export_osmnames/functions.sql

```diff
@@ -43,47 +43,52 @@ END;
 $$ LANGUAGE plpgsql IMMUTABLE;
 
 
-DROP FUNCTION IF EXISTS get_parent_info(TEXT, BIGINT, INTEGER);
-CREATE FUNCTION get_parent_info(display_name TEXT, polygon_id BIGINT, current_rank INTEGER) RETURNS parentInfo AS $$
+DROP FUNCTION IF EXISTS get_parent_info(TEXT, BIGINT, BIGINT, BOOLEAN, TEXT);
+CREATE FUNCTION get_parent_info(name TEXT, id BIGINT, parent_id BIGINT, is_polygon BOOLEAN, type TEXT) RETURNS parentInfo AS $$
 DECLARE
-  retVal parentInfo;
+  retval parentInfo;
   current_name TEXT;
-  current_parent_id BIGINT;
+  current_rank INTEGER;
+  current_id BIGINT;
+  city_rank INTEGER := 16;
+  county_rank INTEGER := 10;
 BEGIN
-  current_name := display_name;
-  retVal.displayName := current_name;
-
-  IF current_rank BETWEEN 16 AND 20 THEN
-    retVal.city := retVal.displayName;
-  ELSIF current_rank BETWEEN 12 AND 15 THEN
-    retVal.county := retVal.displayName;
-  ELSIF current_rank BETWEEN 8 AND 11 THEN
-    retVal.state := retVal.displayName;
-  END IF;
 
-  current_parent_id := polygon_id;
-  WHILE current_rank >= 8 AND current_parent_id IS NOT NULL LOOP
-    SELECT
-      name,
-      place_rank,
-      parent_id
-    FROM osm_polygon
-    WHERE id = current_parent_id
-    INTO current_name, current_rank, current_parent_id;
-
-    IF current_name IS NOT NULL THEN
-      retVal.displayName := retVal.displayName || ', ' || current_name;
+  IF is_polygon AND type NOT IN ('water', 'bay', 'desert', 'reservoir') THEN
+    current_id := id;
+    retval.displayName := '';
```

Comment:
Yeap, to avoid the `name` from appearing twice in the `displayName` since if this condition is true we set the `current_id` as the element's id. In first iteration of the WHILE loop, this element's name would be SELECTed as the `current_name` (line [66](https://github.com/simran001/OSMNames/blob/master/osmnames/export_osmnames/functions.sql#L66) to 69) which would be assigned to the `displayName` (line [72](https://github.com/simran001/OSMNames/blob/master/osmnames/export_osmnames/functions.sql#L72)). For the following iterations, `current_name` would be the element's parent's name and concatenated with the existing `displayName` (line [74](https://github.com/simran001/OSMNames/blob/master/osmnames/export_osmnames/functions.sql#L74))